### PR TITLE
Issue #20633: Execute Servlet 3.1 FAT with Jakarta EE10 features

### LINKED ISSUE #20633: Servlet 6.0 - additional : Repeat com.ibm.ws.webcontainer.servlet.3.1_fat
Repeat to run on Servlet 6.0

com.ibm.ws.webcontainer.servlet.3.1_fat

### PR DESCRIPTION
**Pull request by pnicolucci**
fixes #20633

### CODE REVIEW

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#libby #build

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/FATSuite.java [hunk 1]
```diff
@@ -85,8 +81,8 @@ public static void generateHelpFile() {
...
     public static RepeatTests r = RepeatTests
                     .with(new EmptyAction().fullFATOnly())
-                    .andWith(FeatureReplacementAction.EE8_FEATURES()
-                                    .fullFATOnly())
-                    .andWith(FeatureReplacementAction.EE9_FEATURES()); 
-                    
+                    .andWith(FeatureReplacementAction.EE8_FEATURES().fullFATOnly())
+                    .andWith(FeatureReplacementAction.EE9_FEATURES().fullFATOnly())
+                    .andWith(FeatureReplacementAction.EE10_FEATURES());
+
```
Comment (pmd1nh) [inline dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/FATSuite.java:87]: should EE10_FEATURES() also be fullFATOnly() ?

Comment (volosied) [reply dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/FATSuite.java:87]: No, Paul's code looks right. Only the EE10_FEATURES features would run in lite mode. When full mode rules, then the the original, ee8, ee9, and ee10 versions would all run.  

For the example, the build showed that 
Lite Mode: 99 tests ran
Full Mode: 577 tests ran

**Review by volosied — APPROVED** [review]
Looks good as far as I can tell.

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/FATSuite.java (hunk 1 above)
Comment (pnicolucci) [reply dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/FATSuite.java:87]: https://github.com/OpenLiberty/open-liberty/wiki/FAT-Tests#repeating-fat-tests-for-different-featurejee-levels : 

> It is recommended that only the latest version of a feature should be tested in LITE mode. Other versions should only be repeated in FULL mode.

**Review by pmd1nh — APPROVED** [review]

**Pull request merged**